docs/writing-tests/test-runner.md

```diff
@@ -0,0 +1,293 @@
+Listed below are the available hooks and an overview of how to use them.
+
+| Hook         | Description                                                                   |
+| ------------ | ----------------------------------------------------------------------------- |
+| `setup`      | Executes once before all the tests run<br/>`setup() {}`                       |
+| `preRender`  | Executes before a story is rendered<br/>`async preRender(page, context) {}`   |
+| `postRender` | Executes after the story is rendered<br/>`async postRender(page, context) {}` |
+
+<div class="aside">
+
+💡 These test hooks are experimental and may be subject to breaking changes. We encourage you to test as much as possible within the story's [play function](../writing-stories/play-function.md).
+
+</div>
+
+To enable the hooks API, you'll need to add a new configuration file inside your Storybook directory and set them up as follows:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-test-runner-hooks-example.js.mdx',
+    'common/storybook-test-runner-hooks-example.ts.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<div class="aside">
+
+💡 Except for the `setup` function, all other functions run asynchronously. Both `preRender` and `postRender` functions include two additional arguments, a [Playwright page](https://playwright.dev/docs/pages) and a context object which contains the `id`, `title`, and the `name` of the story.
+
+</div>
+
+When the test runner executes, your existing tests will go through the following lifecycle:
+
+- The `setup` function is executed before all the tests run.
+- The context object is generated containing the required information.
+- Playwright navigates to the story's page.
+- The `preRender` function is executed.
+- The story is rendered, and any existing `play` functions are executed.
+- The `postRender` function is executed.
+
+### Stories.json mode
+
+When testing a local Storybook, the test runner transforms your story files into tests. For a remote Storybook, it uses the Storybook's [stories.json](../configure/overview.md#feature-flags) file (a static index of all the stories) to run the tests. Suppose you run into a situation where the local and remote Storybooks appear out of sync, or you might not even have access to the code. In that case, the `stories.json` file is guaranteed to be the most accurate representation of the deployed Storybook you are testing. To test a local Storybook using this feature, use the `--stories-json` flag as follows:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-test-runner-with-stories-json.yarn.js.mdx',
+    'common/storybook-test-runner-with-stories-json.npm.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<div class="aside">
+
+💡 The `stories.json` mode is not compatible with watch mode.
+
+</div>
+
+To disable it, use the `--no-stories-json` flag:
```

Comment:
Why would someone need to disable it? The README includes this:

> By default, the test runner will detect whether your Storybook URL is local or remote, and if it is remote, it will run in "stories.json mode" automatically. To disable it, you can pass the --no-stories-json flag

Reply:
@yannbf you explained this to me but I forget.

Reply:
Sorry, I should've been more clear. I think the README explains the "why" well, and this doc should do something similar.

Reply:
@kylegach to give you a bit more context on this, I'm aware of it and left it as is intentionally,  as the idea was to spin up a page that details all the existing feature flags as what's currently documented needs some additional work, to which the people I've talked with were on board with. The perfect example of this is in the notion doc that @winkerVSbecks shared with you in a previous comment. Once I have it materialized I would love your feedback on it.